extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -79,7 +79,8 @@ public SecurityIdentity get() {
 
         if (request.getToken() instanceof IdTokenCredential
                 && (resolvedContext.oidcConfig.authentication.verifyAccessToken
-                        || resolvedContext.oidcConfig.roles.source.get() == Source.accesstoken)) {
+                        || resolvedContext.oidcConfig.roles.source.isPresent()
+                                && resolvedContext.oidcConfig.roles.source.get() == Source.accesstoken)) {
```

Comment:
```suggestion
                        || (resolvedContext.oidcConfig.roles.source.isPresent()
                                && resolvedContext.oidcConfig.roles.source.get() == Source.accesstoken))) {
```

Reply:
You can also write
```
resolvedContext.oidcConfig.roles.source.orElse(null) == Source.accesstoken
```
if you want to simplify it :wink:

Reply:
HI @gastaldi That is not a bug AFAIK, the `&&` expression takes priority over `||`, agreed ?

Reply:
HI @gastaldi That is not a bug AFAIK, the `&&` expression takes priority over `||`, agreed ?

Reply:
@gastaldi but I'll just commit `resolvedContext.oidcConfig.roles.source.orElse(null) == Source.accesstoken` though as it will simplify the expression :-)

Reply:
> HI @gastaldi That is not a bug AFAIK, the `&&` expression takes priority over `||`, agreed ?

Ah yes, I forgot this minor detail, but the actual code looks better now 😉